### ISSUE TITLE
standardizeCycleTime discussion

### ISSUE THREAD

**Issue by masterl** [masterl/MaintenanceMadnessReloaded#15]
This is the original code:

```lua
function mm_util.standardizeCycleTime( args )
    local MTBM = args.MTBM -- needs to be in ticks
    local MTTM = args.MTTM -- needs to be in ticks
    local MTTR = args.MTTR -- needs to be in ticks
    -- local replacementAge = args.replacementAge -- percentage of life time, 80 by default
    local maxAge = args.maxAge -- index value, 100 by default
    local repairTimeModifier = args.repairTimeModifier
    local repairProbabilityModifier = args.repairProbabilityModifier
    local lifeTime = args.lifeTime -- needs to be in ticks

    local avgAge = maxAge / 2
    local avgMalfunctionRatio = math.min(
                                    (avgAge / maxAge) *
                                        repairProbabilityModifier, 1 )
    local avgMaintenanceRatio = 1 - avgMalfunctionRatio
    local avgMalfunctionTime = MTTR +
                                   (MTTR * repairTimeModifier *
                                       (avgAge / maxAge))
    local avgCycleTime =
        MTBM + MTTM * avgMaintenanceRatio + avgMalfunctionTime *
            avgMalfunctionRatio

    -- one cycle is the time from the start of one maintenance event to the start of the next event, by default about 13 min
    local expectedCyclesPerLifeTime = lifeTime / avgCycleTime

    return expectedCyclesPerLifeTime
end
```

As written, it's pretty obvious that (avgAge / maxAge) would always be 0.5, so we could rewrite this as:

```lua
function mm_util.standardizeCycleTime( args )
    local MTBM = args.MTBM -- needs to be in ticks
    local MTTM = args.MTTM -- needs to be in ticks
    local MTTR = args.MTTR -- needs to be in ticks
    -- local replacementAge = args.replacementAge -- percentage of life time, 80 by default
    local repairTimeModifier = args.repairTimeModifier
    local repairProbabilityModifier = args.repairProbabilityModifier
    local lifeTime = args.lifeTime -- needs to be in ticks

    local avgMalfunctionRatio = math.min( 0.5 * repairProbabilityModifier, 1 )
    local avgMaintenanceRatio = 1 - avgMalfunctionRatio
    local avgMalfunctionTime = MTTR + (MTTR * repairTimeModifier * 0.5)
    local avgCycleTime =
        MTBM + MTTM * avgMaintenanceRatio + avgMalfunctionTime *
            avgMalfunctionRatio

    -- one cycle is the time from the start of one maintenance event to the start of the next event, by default about 13 min
    local expectedCyclesPerLifeTime = lifeTime / avgCycleTime

    return expectedCyclesPerLifeTime
end
```

`MTBM` being mean time between maintenance events
`MTTM` being mean time to maintain 
`MTTR` being mean time to repair
`repairTimeModifier` is currently `1.5`
`repairProbabilityModifier` is currently `0.5`

`lifeTime` is currently:
```
hours_to_ticks(
    get_setting_value(
        'expected-life-time' ) ) 
```

`'expected-life-time'` has the default value of `12` (meaning 12 hours)
So basically, `lifeTime` is 12 hours converted to the amount of game ticks

**Comment by masterl** [comment]
So we end up with this:

```lua
    local MTBM = minutes_to_ticks( 12 )
    local MTTM = minutes_to_ticks( 3 )
    local MTTR = minutes_to_ticks( 5 )

    local repairTimeModifier = 1.5
    local repairProbabilityModifier = 0.5
    local lifeTime = hours_to_ticks( 12 )
```

We can then substitute, and this will be what's being calculated:

```lua
    local avgMalfunctionRatio = math.min( 0.5 * 0.5, 1 ) -- Always 0.25
    local avgMaintenanceRatio = 1 - 0.25 -- so: 0.75
    local avgMalfunctionTime = MTTR + (MTTR * 1.5 * 0.5) -- 1.75 * MTTR
    local avgCycleTime = MTBM + MTTM * 0.75 + 1.75 * MTTR * 0.25 -- MTBM + MTTM * 0.75 + MTTR * 0.4375

    -- one cycle is the time from the start of one maintenance event to the start of the next event, by default about 13 min
    local expectedCyclesPerLifeTime = lifeTime / avgCycleTime

    return expectedCyclesPerLifeTime
```

So, considering default values, the average cycle time ends up being:
- Mean time between maintenance events
- :heavy_plus_sign: 75% of the mean time to maintain
- :heavy_plus_sign: 43.75% of the mean time to repair

And the last step is dividing the expected life time of a machine by this time we calculated.

So this function is basically estimating how many maintenance cycles a machine will have during it's life, as pointed out by it's result variable name.
**But the function name itself isn't good since it isn't "standardizing" anything.**

**Comment by masterl** [comment]
To summarize this issue, I'm just not sure if this is the best approach to averaging the cycle time:

- Mean time between maintenance events
- :heavy_plus_sign: 75% of the mean time to maintain
- :heavy_plus_sign: 43.75% of the mean time to repair

I think I need to explore `control.lua` and `control-gui.lua` before changing this formula.
